service_configuration_lib/spark_config.py

```diff
@@ -46,7 +47,13 @@ def get_mesos_spark_auth_env() -> Mapping[str, str]:
 
 def _check_non_configurable_spark_opts(user_spark_opts: Mapping[str, Any]) -> None:
     user_non_config_opts = set(user_spark_opts) & NON_CONFIGURABLE_SPARK_OPTS
-    if user_non_config_opts:
+    if (
+        user_non_config_opts or
+        any([
+            spark_opt.startswith('spark.kubernetes.executor.volumes.hostPath')
```

Comment:
why not just add these to the set of non-configurable opts?

Reply:
Because these opts are dynamically generated therefore cannot be statically compared like the rest opts.